pymeasure/adapters/visa.py

```diff
@@ -71,34 +76,35 @@ def has_supported_version():
             return False
 
     @staticmethod
-    def setdefaults(resource_name, interface_type, user_kwargs, **kwargs):
-        """ Change user_kwargs to add defaults specified by user for the selected interface type
+    def setdefaults(resource_name, received_kwargs, resource_mgr=None, visa_library=None):
+        """ Remove invalid entries from the received keyword arguments.
 
-        :param resource_name: VISA resource name that identifies the address
-        :param interface_type: VISA interface type for which the defaults should be set
-        :param user_kwargs: keywords parameters dictionary passed by user, this dictionary can be modified by this method.
-        :param **kwargs: keywords argument to be set as default
+        Any entries for the interface expected based on resource_name are added
+        to received_kwargs.
+        All interface-specific entries are removed as they are not expected by pyvisa as-is.
 
+        :param resource_name: VISA resource name that identifies the address
+        :param received_kwargs: keywords parameters dictionary to be cleaned in-place.
+        :param resource_mgr: A pyvisa resource manager. Will be created if not given.
+        :param visa_library: Optional pyvisa visa_library specifier for creating the resource manager.
         """
         # Get VISA interface type associated with resource_name
-        if isinstance(resource_name, int):
-            resource_name = "GPIB0::%d::INSTR" % resource_name
-        elif isinstance(resource_name, str):
-            pass
-        else:
-            # User defined adapter, do not touch anything
-            return
```

Comment:
this can be removed because it duplicates what already happens in VISAAdapter right before this method is called.